modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder_render_rule_condition.js

```diff
@@ -497,7 +497,13 @@ AUI.add(
 			_isConstant: function(operandTypeValue) {
 				var instance = this;
 
-				return operandTypeValue === 'double' || operandTypeValue === 'integer' || operandTypeValue === 'string';
+				return operandTypeValue === 'double' || operandTypeValue === 'integer' || operandTypeValue === 'string' || operandTypeValue === 'date';
+			},
+
+			_isDate: function(operandTypeValue) {
+				var instance = this;
+
+				return operandTypeValue === 'date';
```

Comment:
Here too,If you don't want to compare the type of the data, you just need to use two equals (==), because is more performative. But let me know your thinking.